runner/run_test.go

```diff
@@ -410,37 +411,54 @@ func TestRun(t *testing.T) {
 	replaceDestinationInTest(&ftwTest, *dest)
 
 	t.Run("show time and execute all", func(t *testing.T) {
-		if res := Run("", "", true, false, []test.FTWTest{ftwTest}); res.Stats.TotalFailed() > 0 {
+		if res := Run([]test.FTWTest{ftwTest}, Config{
+			ShowTime: true,
+			Quiet:    true,
+		}); res.Stats.TotalFailed() > 0 {
 			t.Errorf("Oops, %d tests failed to run!", res.Stats.TotalFailed())
 		}
 	})
 
 	t.Run("be verbose and execute all", func(t *testing.T) {
-		if res := Run("0*", "", true, true, []test.FTWTest{ftwTest}); res.Stats.TotalFailed() > 0 {
+		if res := Run([]test.FTWTest{ftwTest}, Config{
+			Include:  regexp.MustCompile("0*"),
+			ShowTime: true,
+		}); res.Stats.TotalFailed() > 0 {
 			t.Error("Oops, test run failed!")
 		}
 	})
 
 	t.Run("don't show time and execute all", func(t *testing.T) {
-		if res := Run("0*", "", false, false, []test.FTWTest{ftwTest}); res.Stats.TotalFailed() > 0 {
+		if res := Run([]test.FTWTest{ftwTest}, Config{
+			Include: regexp.MustCompile("0*"),
+		}); res.Stats.TotalFailed() > 0 {
 			t.Error("Oops, test run failed!")
 		}
 	})
 
 	t.Run("execute only test 008 but exclude all", func(t *testing.T) {
-		if res := Run("008", "0*", false, false, []test.FTWTest{ftwTest}); res.Stats.TotalFailed() > 0 {
+		if res := Run([]test.FTWTest{ftwTest}, Config{
+			Include: regexp.MustCompile("008"),
+			Exclude: regexp.MustCompile("0*"),
+		}); res.Stats.TotalFailed() > 0 {
 			t.Error("Oops, test run failed!")
 		}
 	})
 
 	t.Run("exclude test 010", func(t *testing.T) {
-		if res := Run("*", "010", false, false, []test.FTWTest{ftwTest}); res.Stats.TotalFailed() > 0 {
```

Comment:
There were some issues with the tests using `"*"`, which isn't a valid regexp so was hitting the `err` branch of the matchers. Changing the programmatic way of running to accept `*regexp.Regexp` directly allows users to see such errors in the IDE, and for go-ftw to not need to worry about problematic regexps from user

Reply:
Excellent. I think I saw this behavior before, so a welcomed change.